### PR TITLE
Keyword `title`

### PR DESCRIPTION
**Pull request by AgniveshChaubey**
This PR addresses #71.

Please have a look at the last example and let me know whether I understood correctly what you meant to convey in issue #71.

Regarding the output format for annotations, I've maintained it as per the discussion in issue #133. Although you previously suggested replacing the ellipsis used in the annotation with the complete standard output, I haven't replaced it as the complete output would be too large. Please let me know your thoughts on this.

### CODE REVIEW

File: content/2020-12/meta-data/title.markdown [hunk 1]
```diff
@@ -27,3 +27,165 @@ Annotations
...
+{{</schema>}}
+
+{{<instance-pass `An instance with a numeric value is valid`>}}
+45
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+{
+  "valid": true,
+  "keywordLocation": "/",
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:52]:
```suggestion
  "keywordLocation": "/title",
```

The keyword emitting the annotation is `title`

File: content/2020-12/meta-data/title.markdown [hunk 2]
```diff
@@ -27,3 +27,165 @@ Annotations
...
+  "instanceLocation": "",
+  "annotations": {
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:54]:
```suggestion
  "annotations": "Age of a person"
```

And the annotation is the title itself

File: content/2020-12/meta-data/title.markdown [hunk 3]
```diff
@@ -27,3 +27,165 @@ Annotations
...
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  /// ...
+  {
+    "valid": true,
+    "keywordLocation": "/",
+    "instanceLocation": "",
+    "annotations": {
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:104]: This annotation example is mixing up annotations from different keywords. Each keyword generates a separate annotation. So you would have one string annotation by the `/title` keyword, one annotation by the `properties` keyword, etc

File: content/2020-12/meta-data/title.markdown [hunk 4]
```diff
@@ -27,3 +27,165 @@ Annotations
...
+  "eligible": true
+}
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  /// ...
+  {
+    "valid": true,
+    "keywordLocation": "/",
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:102]: Also, this pointer is here is invalid. `/` means "the empty property". For the top-level object, the correct pointer is just `""`

In any case, this whole thing is invalid, as the root of the schema itself is not producing the annotations. Each of the corresponding keywords is, with the corresponding `keywordLocation`

File: content/2020-12/meta-data/title.markdown [hunk 5]
```diff
@@ -27,3 +27,165 @@ Annotations
...
+    "instanceLocation": "",
+    "annotations": {
+      "title": "Personal Info",
+      "properties": [ "name", "age" ],
+      "if": true
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:107]: I don't think `if` produces an annotation?

**Comment by jviotti** [comment]
The last example is good, but the annotation outputs require correction. In many of these, you have a single output unit from a non-keyword (like `""`) producing multiple annotations for different keywords. We want each individual keyword to produce an annotation of its own as a separate output unit

File: content/2020-12/meta-data/title.markdown (hunk 5 above)
Comment (AgniveshChaubey) [reply content/2020-12/meta-data/title.markdown:107]: I too was confused with it. This annotation result was produced by json-everything and I intentionally kept it to get your feedback on this.

**Comment by AgniveshChaubey** [comment]
Updated the annotation results!

File: content/2020-12/meta-data/title.markdown (hunk 5 above)
Comment (jviotti) [reply content/2020-12/meta-data/title.markdown]: Yeah, doesn't seem to be standard according to https://json-schema.org/draft/2020-12/json-schema-core#section-10.2.2.1. Let's remove it!

File: content/2020-12/meta-data/title.markdown [hunk 6]
```diff
@@ -27,3 +27,159 @@ Annotations
...
+{{<instance-pass `An instance with a numeric value is valid`>}}
+45
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+{
+  "valid": true,
+  "keywordLocation": "/title",
+  "instanceLocation": "",
+  "annotations": "Age of a person"
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:54]: This should be just `annotation`

File: content/2020-12/meta-data/title.markdown [hunk 7]
```diff
@@ -27,3 +27,159 @@ Annotations
...
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  /// ...
+  {
+    "valid": true,
+    "keywordLocation": "/title",
+    "instanceLocation": "",
+    "annotations": "Personal Info"
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:102]: Same in all of these. Should be `annotation` (in singular form)

File: content/2020-12/meta-data/title.markdown [hunk 8]
```diff
@@ -27,3 +27,159 @@ Annotations
...
+  },
+  {
+    "valid": true,
+    "keywordLocation": "/properties",
```
Comment (jviotti) [inline content/2020-12/meta-data/title.markdown:106]: While this is correct, maybe let's use ellipses to "hide" any annotation that is not relevant to the keywords we are describing? i.e. here we can only show annotations generated by `title`?

Comment (AgniveshChaubey) [reply content/2020-12/meta-data/title.markdown:106]: Sure!

**Comment by AgniveshChaubey** [comment]
Updated!

**Review by jviotti — APPROVED** [review]
Awesome!

**Pull request merged**